### PR TITLE
Move supported drivers per platform

### PR DESCRIPTION
**Pull request by josedonizetti**
This is something `minishift` do and is pretty useful. Wdyt?

### CODE REVIEW

**Comment by afbjorklund** [comment]
To be honest I'm not sure why even bother with plugins, if we are hard-coding the list of "supported" ones anyway ? It's not like you can add your own, since `minikube' will just complain that it is not supported...

But since we are going the way of validating the drivers (#4720), then maybe this makes the code prettier/smaller. And first glance it just looks like duplication, but maybe that can be cleaned up a bit ?

**Comment by josedonizetti** [comment]
@afbjorklund I don't see it the same way. I understood this `supported` drivers as something for the documentation, maybe the naming isn't good, but it is simple to change. I think the documentation should be as clean as possible, listing the KVM driver for a Windows user when he runs `minikube start --help` is misleading, and the idea here is to improve this documentation, not to force the use of only those drivers. That's the case for the PR https://github.com/kubernetes/minikube/pull/4720 as well, it should only validate that if someone tries the none driver in Darwin or Windows he is alerted this won't work, cos it's not for those environments, not that only those drivers are now supported and no one can try to create his one using a plugin.

**Comment by afbjorklund** [comment]
I’m all for filtering out the invalid drivers (i.e. not available on platform)

File: pkg/minikube/constants/constants_darwin.go [hunk 1]
```diff
@@ -19,3 +19,13 @@ limitations under the License.
...
 
 var DefaultMountDir = "/Users"
+
+// SupportedVMDrivers is a list of supported drivers on Darwin.
+var SupportedVMDrivers = [...]string{
+	DriverVirtualbox,
+	DriverParallels,
+	DriverVmwareFusion,
+	DriverXhyve,
+	DriverHyperv,
```
Comment (tstromberg) [inline pkg/minikube/constants/constants_darwin.go]: This should be HyperKit, not HyperV.

Comment (josedonizetti) [reply pkg/minikube/constants/constants_darwin.go]: great catch!

File: pkg/minikube/constants/constants_windows.go [hunk 2]
```diff
@@ -23,3 +23,12 @@ import (
 )
 
 var DefaultMountDir = homedir.HomeDir()
+
+// SupportedVMDrivers is a list of supported drivers on Windows.
+var SupportedVMDrivers = [...]string{
+	DriverVirtualbox,
+	DriverParallels,
```
Comment (blueelvis) [inline pkg/minikube/constants/constants_windows.go]: I am pretty sure that Parallels is not supported/can be used on Windows. Pretty sure it is only for Mac. Could you please confirm?

Comment (josedonizetti) [reply pkg/minikube/constants/constants_windows.go]: Thanks!

**Comment by afbjorklund** [comment]
Apparently it got cancelled. https://www.parallels.com/blogs/parallels-desktop-for-windows/

**Pull request merged**